### PR TITLE
Use layer max zoom levels + layer name change

### PR DESCRIPTION
**Pull request by caseypt**
Uses the maximum zoom level of each layer instead of a fixed value for all. Also changes the layer names to be generic and adds the Google Terrain layer.

See commit messages for details.

**Testing different maximum zoom level**
- Keep the default basemap.
- Center the map on City Hall.
- Switch to the "Satellite" basemap.
- You should be able to zoom one level closer.
- Switch back to the "Streets" basemap.
- You should be zoomed out one level further than before, and zooming in should not be available.
- Switch to the "Satelitte with Streets" basemap.
- You should be able to zoom in two levels.
- Switch back to the "Streets" basemap.
- You should be zoomed out two levels further than before, and zooming in should be not be available.

**Testing base layer name changes**
- You should see non-branded names in the layer selector, like so:

screen shot 2015-08-18 at 1 59 48 pm

Connects to #692 
Connects to #696

### CODE REVIEW

**Comment by caseypt** [comment]
Rebased on `develop`.

**Comment by caseypt** [comment]
Tests fixed.

**Comment by rajadain** [comment]
New sessions start out with the "Streets" basemap at max zoom to my current location, as opposed to the current behavior:

image
image

Is this desirable?

**Comment by caseypt** [comment]
That behavior is because we set the `maxZoom` to 10 when we first added the map: https://github.com/WikiWatershed/model-my-watershed/commit/330b5542a1301791cfcc715f70e0b7d926d4b308.

It's now set to the actual max zoom of the layer.

I don't have a strong preference on where it should be set, so happy to defer to others.

I do think we should get rid of this line though or modify it to center on the US: https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/js/src/core/views.js#L240

File: src/mmw/js/src/core/views.js [hunk 1]
```diff
@@ -461,12 +483,53 @@ var MapView = Marionette.ItemView.extend({
...
+
+        this._googleMaxZoomService.getMaxZoomAtLatLng(latLng, function(response) {
+            if (response.status !== self._googleMaps.MaxZoomStatus.OK) {
+                // Leave the max layer zoom as is
+                return;
+            } else {
+                // Set layer zoom level to the max for the current area
+                _.each(self.baseLayers, function(layer) {
+                    // Needs to be updated if other Google Base maps are added
+                    if (layer._type === 'HYBRID') {
```
Comment (mmcfarland) [inline src/mmw/js/src/core/views.js]: Should the Terrain option be listed here as well?  Maybe all `layer` types that are of Google should get an additional attribute set so this conditional won't need to be updated?  Like `layer.isGoogle`?

Comment (caseypt) [inline src/mmw/js/src/core/views.js]: As far as I can tell from the [Google docs](https://developers.google.com/maps/documentation/javascript/3.exp/reference#MaxZoomService), it's just satellite based layers. Unfortunately, we can't add any custom properties to the Leaflet layer object, but I'll add the other map type that uses imagery.

Comment (mmcfarland) [inline src/mmw/js/src/core/views.js]: Got it - I was just going off of the comment.

**Comment by mmcfarland** [comment]
Re: default zoom and location, I agree, I think a default of regional scale - similar to where it was originally is a pretty good level.  I don't think eastern PA is a good starting point either.  Somewhere central seems good, and then most people will geolocate to their location.

File: src/mmw/js/src/core/views.js (hunk 1 above)
Comment (caseypt) [inline src/mmw/js/src/core/views.js]: Not a great comment, considering adding support for the other layer was less characters.

**Comment by caseypt** [comment]
Update with support for both satellite-based Google layers, as well as new setting for the default extent and new zoom level for geolocating.

Default extent:

screen shot 2015-08-19 at 3 11 57 pm

Geolocation zoom level:

screen shot 2015-08-19 at 3 11 25 pm

**Comment by mmcfarland** [comment]
Looks good.  Let's hope it doesn't encourage them to turn on HUC12 at that scale...

**Comment by rajadain** [comment]
+1

**Pull request merged**